packages/ui/src/DataTable.tsx

```diff
@@ -0,0 +1,698 @@
+          borderRightColor: theme.border.default,
+        }),
+      }}
+    >
+      <Component cellData={{...value, textSize}} column={columnDef} />
+    </View>
+  );
+};
+
+interface DataTableRowProps {
+  rowData: CellData[];
+  rowIndex: number;
+  columns: DataTableColumn[];
+  pinnedColumns: number;
+  columnWidths: number[];
+  alternateRowBackground: boolean;
+  customColumnComponentMap?: CustomComponentMap;
+  rowHeight: number;
+}
+
+const DataTableRow: React.FC<DataTableRowProps> = ({
+  rowData,
+  rowIndex,
+  columns,
+  pinnedColumns,
+  columnWidths,
+  alternateRowBackground,
+  customColumnComponentMap,
+  rowHeight,
+}) => {
+  const {theme} = useTheme();
+  const backgroundColor =
+    alternateRowBackground && rowIndex % 2 === 1 ? theme.surface.neutralLight : theme.surface.base;
+
+  return (
+    <View
+      style={{
+        flexDirection: "row",
+        height: rowHeight,
+        borderBottomWidth: 1,
+        borderBottomColor: theme.border.default,
+      }}
+    >
+      {rowData.map((cell, colIndex) => (
+        <DataTableCell
+          key={colIndex}
+          backgroundColor={
+            cell.highlight ? theme.surface[cell.highlight as SurfaceColor] : backgroundColor
+          }
+          colIndex={colIndex}
+          columnDef={columns[colIndex]}
+          columnWidths={columnWidths}
+          customColumnComponentMap={customColumnComponentMap}
+          height={rowHeight}
+          isPinned={colIndex < pinnedColumns}
+          pinnedColumns={pinnedColumns}
+          textSize={cell.textSize}
+          value={cell}
+        />
+      ))}
+    </View>
+  );
+};
+
+interface MoreButtonCellProps {
+  rowIndex: number;
+  alternateRowBackground: boolean;
+  onClick: (rowIndex: number) => void;
+  column: DataTableColumn;
+  rowHeight: number;
+}
+
+const MoreButtonCell: React.FC<MoreButtonCellProps> = ({
+  rowIndex,
+  alternateRowBackground,
+  onClick,
+  rowHeight,
+}) => {
+  const {theme} = useTheme();
+
+  return (
+    <View
+      style={{
+        width: 48,
+        height: rowHeight ?? 54,
+        justifyContent: "center",
+        alignItems: "center",
+        backgroundColor:
+          alternateRowBackground && rowIndex % 2 === 1
+            ? theme.surface.neutralLight
+            : theme.surface.base,
+        borderBottomWidth: 1,
+        borderBottomColor: theme.border.default,
+      }}
+    >
+      <IconButton
+        accessibilityHint="View details"
+        accessibilityLabel="Open modal"
+        iconName="info"
+        variant="secondary"
+        onClick={() => onClick(rowIndex)}
+      />
+    </View>
+  );
+};
+
+interface DataTableHeaderCellProps {
+  column: DataTableColumn;
+  index: number;
+  isPinned: boolean;
+  columnWidths: number[];
+  sortColumn?: ColumnSortInterface;
+  onSort: (index: number) => void;
+  rowHeight: number;
+}
+
+const DataTableHeaderCell: React.FC<DataTableHeaderCellProps> = ({
+  column,
+  index,
+  isPinned,
```

Comment:
Is there situation where we wouldn't want to pin the header?

Reply:
Yes! If the column is pinned, the first header cell is pinned (horizontally), so when you scroll, e.g. "Name" stays above the Name column. But that is confusing with the naming, I'll update.